sys/log/full/include/log/log.h

```diff
@@ -218,6 +272,21 @@ struct log {
 #endif
 #if MYNEWT_VAL(LOG_STATS)
     STATS_SECT_DECL(logs) l_stats;
+#endif
+    /* Custom log init callback to be called by the last hdr
+     * read function to read custom data from log entries
+     * at init
+     */
+    log_walk_func_t l_init_cb;
+
+#if MYNEWT_VAL(LOG_FLAGS_TRAILER)
+    /* Log trailer support */
+    /* Void argument for callbacks registeration */
+    void *l_tr_arg;
```

Comment:
this argument shall be passed to all callbacks, otherwise it doesn't make any sense

Reply:
done